src/rgw/services/svc_notify.cc

```diff
@@ -360,11 +360,14 @@ void RGWSI_Notify::_set_enabled(bool status)
 int RGWSI_Notify::distribute(const string& key, bufferlist& bl,
                              optional_yield y)
 {
-  RGWSI_RADOS::Obj notify_obj = pick_control_obj(key);
+  if (num_watchers > 0) {
+    RGWSI_RADOS::Obj notify_obj = pick_control_obj(key);
 
-  ldout(cct, 10) << "distributing notification oid=" << notify_obj.get_ref().obj
-      << " bl.length()=" << bl.length() << dendl;
-  return robust_notify(notify_obj, bl, y);
+    ldout(cct, 10) << "distributing notification oid=" << notify_obj.get_ref().obj
+        << " bl.length()=" << bl.length() << dendl;
+    return robust_notify(notify_obj, bl, y);
+  }
+  return -ENOENT;
```

Comment:
why is this part changing?

Reply:
without it. it will try to distribute the cache even without notify RADOS objects (on cluster init), which will cause divide by zero in RGWSI_RADOS::Obj RGWSI_Notify::pick_control_obj (num_watchers is zero)

Reply:
is there any way to produce this other than setting rgw_num_control_oids=0? i think we're better off requiring a non-zero number:
```diff
     Option("rgw_num_control_oids", Option::TYPE_INT, Option::LEVEL_ADVANCED)
     .set_default(8)
+  .set_min(1)
```

Reply:
@cbodley it will not solve the issue, the issue is that on cluster init we are calling the end the pick_control_obj although we haven't run yet init_watch() to create

Reply:
can you show where this is being called before it's initialized?